qiskit/_compiler.py

```diff
@@ -326,6 +322,52 @@ def best_subset(backend, n_qubits):
     return best_map
 
 
+def _matches_coupling_map(instructions, coupling_map):
+    """ Iterate over circuit instructions set to check if multi-qubit instructions coupling
+        graphs are equal to backend qubits coupling graph.
+
+    Parameters:
+            instructions (List): List of circuit instructions. Noy all instructions
+                                 are Gates, hence not multi-qubit.
+            coupling_map (List): Backend coupling map, represented as an
+                                          adjacency list.
+
+    Returns:
+            True: If there's at least one instruction that uses multiple qubits and
+                  these qubits coupling matches one of the backend couplings.
+
+            False: If there's no match between any of the instructions multiple qubits
+                   coupling, and one of the couplings from the backend.
+    """
+    for instruction in instructions:
+        if isinstance(instruction, Gate) and instruction.is_multi_qubit():
+            if instruction.get_qubit_coupling() not in coupling_map:
```

Comment:
For a two qubit gate between qubits `i` and `j`, I thought the check would be if `j` is in `coupling_map[i]` or if `i` in `coupling_map[j]`.  Is that what is going on here, or (more likely) am I missing something?

Reply:
This checks for exact match, meaning if there is a CX(q[i], q[j]) in the circuit, it checks that the device also has CX(q[i], q[j]) in it. The reason is that I want to keep already-matching circuits untouched, but pass any other ones through the mapper. A good mapper may actually find a more compact circuit, but usually a user that writes an already-matching circuit wants to use those exact qubits.

Reply:
Yeah, I found this confusing at first, because I thought that the `coupling_map` was an adjacency list, but it's not. It more a list of coupled pairs: `coupling_map = [[1, 0], [2, 0], [2, 1], [3, 2], [3, 4], [4, 2]]`.

Reply:
Yes, your right.  This is what we want.